src/mongoc/mongoc-server-description-private.h

```diff
@@ -28,19 +28,9 @@
 /* represent a server or topology with no replica set config version */
 #define MONGOC_NO_SET_VERSION -1
 
-typedef enum
-   {
-      MONGOC_SERVER_UNKNOWN,
-      MONGOC_SERVER_STANDALONE,
-      MONGOC_SERVER_MONGOS,
-      MONGOC_SERVER_POSSIBLE_PRIMARY,
-      MONGOC_SERVER_RS_PRIMARY,
-      MONGOC_SERVER_RS_SECONDARY,
-      MONGOC_SERVER_RS_ARBITER,
-      MONGOC_SERVER_RS_OTHER,
-      MONGOC_SERVER_RS_GHOST,
-      MONGOC_SERVER_DESCRIPTION_TYPES,
-   } mongoc_server_description_type_t;
+/* this is used for declaring gSDAMTransitionTable and should be defined as the
+ * number of unique states in mongoc_server_description_type_t */
+#define MONGOC_SERVER_DESCRIPTION_TYPES 9
```

Comment:
It seems weird to define an enum in one place, the count of it in another place, and then its only used in the 3rd place.

Its unfortunate that this enum must be exposed, because it means we are knowingly breaking the ABI as soon as we add another entry to it, which will break your "enum to string" right off the bat.

I strongly recommend instead that we rather return the stringification of the types.
If you want to represent that as a number somewhere else, then thats fine. Any unknown strings can then be represented as a 0 in the app or something....